src/components/cc-token-api-creation-form/cc-token-api-creation-form.js

```diff
@@ -0,0 +1,878 @@
+    switch (activeStep) {
+      case 'configuration':
+        return i18n('cc-token-api-creation-form.configuration-step.main-heading');
+      case 'validation':
+        return i18n('cc-token-api-creation-form.validation-step.main-heading');
+      case 'copy':
+        return i18n('cc-token-api-creation-form.copy-step.main-heading');
+    }
+  }
+
+  /**
+   * @param {TokenApiCreationFormStateLoaded['activeStep']} activeStep
+   * @returns {string}
+   */
+  _getDescription(activeStep) {
+    switch (activeStep) {
+      case 'configuration':
+        return i18n('cc-token-api-creation-form.configuration-step.description');
+      case 'validation':
+        return i18n('cc-token-api-creation-form.validation-step.description');
+      case 'copy':
+        return i18n('cc-token-api-creation-form.copy-step.description');
+    }
+  }
+
+  /** @returns {Array<{ label: string, value: ExpirationDuration}>} */
+  _getExpirationDurationOptions() {
+    return [
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.seven-days'),
+        value: 'seven-days',
+      },
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.thirty-days'),
+        value: 'thirty-days',
+      },
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.sixty-days'),
+        value: 'sixty-days',
+      },
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.ninety-days'),
+        value: 'ninety-days',
+      },
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.one-year'),
+        value: 'one-year',
+      },
+      {
+        label: i18n('cc-token-api-creation-form.configuration-step.form.expiration-duration.option-label.custom'),
+        value: 'custom',
+      },
+    ];
+  }
+
+  /** @param {Event} event */
+  _onConfigLinkClick(event) {
+    event.preventDefault();
+    // to make sure the console router doesn't pick this event as a navigation
+    event.stopPropagation();
+    this.state = /** @type {TokenApiCreationFormStateLoaded} */ ({
+      ...this.state,
+      activeStep: 'configuration',
+    });
+  }
+
+  _onConfigFormSubmit() {
+    this.state = {
+      ...this.state,
+      activeStep: 'validation',
+    };
+  }
+
+  _onValidationFormSubmit() {
+    if (this.state.type !== 'loaded' || this.state.activeStep !== 'validation') {
+      return;
+    }
+
+    // clean up potential error messages related to credentials
+    this.state = {
+      ...this.state,
+      credentialsError: null,
+    };
+
+    dispatchCustomEvent(this, 'api-key-create', {
+      name: this.state.values.name,
+      description: this.state.values.description,
+      expirationDate: dateFormatterIso.format(new Date(this.state.values.expirationDate)),
```

Comment:
Should the smart handle formatting instead? (UI = short format, API = ISO format)